wallet/src/de/schildbach/wallet/payments/SendCoinsTaskRunner.kt

```diff
@@ -33,39 +33,86 @@ import org.bouncycastle.crypto.params.KeyParameter
 import org.dash.wallet.common.services.SendPaymentService
 import org.dash.wallet.common.transactions.ByAddressCoinSelector
 import org.slf4j.LoggerFactory
+import java.security.GeneralSecurityException
 import javax.inject.Inject
 
 
 class SendCoinsTaskRunner @Inject constructor(
     private val walletApplication: WalletApplication
 ) : SendPaymentService {
     private val log = LoggerFactory.getLogger(SendCoinsTaskRunner::class.java)
+    val wallet = walletApplication.wallet ?: throw RuntimeException("this method can't be used before creating the wallet")
```

Comment:
There is a chance that `SendCoinsTaskRunner` is injected and initiated before the wallet is created. Also, the wallet can change but this reference here will remain to the old one. It's better to keep this check in the methods, it's fine if it is duplicated.